uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -984,23 +978,24 @@ public static function calculateBalanceChange(
     $isCalculationUnitInHours = $absenceType->isCalculationUnitInHours();
     $dateDeductionService = LeaveDateAmountDeductionFactory::createForAbsenceType($leaveRequest->type_id);
     $contactWorkPatternService = new ContactWorkPatternService();
+    $leaveRequestDayTypes = array_flip(self::buildOptions('from_date_type', 'validate'));
+    $leaveDateDayTypes = self::getLeaveDayTypesForLeaveRequestDates($fromDate, $toDate, $contactId);
 
     foreach ($datePeriod as $date) {
       if($excludeStartAndEndDates) {
         if(in_array($date->format('Y-m-d'), [$fromDate->format('Y-m-d'), $toDate->format('Y-m-d')])) {
           continue;
         }
       }
-      $publicHolidayLeaveRequestExists = self::publicHolidayLeaveRequestExists($contactId, $date);
+
+      $dayType = $leaveDateDayTypes[$date->format('Y-m-d')];
+      $publicHolidayLeaveRequestExists = $dayType == $leaveRequestDayTypes['public_holiday'];
```

Comment:
The `$publicHolidayLeaveRequestExists` name doesn't make sense anymore. What about `$dayIsPublicHoliday`? Or you can simply use `$dayType == $leaveRequestDayTypes['public_holiday']` as the condition for the if